core/tests/test_utils.py

```diff
@@ -2282,6 +2375,74 @@ def assert_same_list_elements(self, phrases, stdout):
 GenericTestBase = AppEngineTestBase
 
 
+class GenericEmailTestBase(GenericTestBase):
+    """Base class for tests requiring email services."""
+
+    def run(self, result=None):
+        """Adds a context swap on top of the test_utils.run() method so that
+        test classes extending GenericEmailTestBase will automatically have
+        a mailgun api key, mailgun domain name and mocked version of
+        send_email_to_recipients().
+        """
+        with self.swap(feconf, 'MAILGUN_API_KEY', 'key'), (
+            self.swap(feconf, 'MAILGUN_DOMAIN_NAME', 'name')), (
+                self.swap(
+                    email_services, 'send_email_to_recipients',
+                    self.email_services_mock.mock_send_email_to_recipients)):
+            super(EmailTestBase, self).run(result=result)
+
+    @classmethod
+    def setUpClass(cls):
+        super(GenericEmailTestBase, cls).setUpClass()
+        cls.email_services_mock = EmailServicesMock()
```

Comment:
Don't need this. The test methods should not need to call the mock directly. Individual test functions may call stuff like self._get_received_email_messages() or similar (see above note about moving the functions into here).